### ISSUE TITLE
Invalid Flag type for TimestampType: 4

### ISSUE THREAD

**Issue by Novemser** [pingcap/tikv-client-java#164]
Known affected branch:
```sql
master
DAGRequestConstruct
```

SQL:
```sql
select  distinct(tp_timestamp)  from full_data_type_table  order by tp_timestamp
```

Exception:
```java
Job aborted due to stage failure: Task 0 in stage 3180.0 failed 1 times, most recent failure: Lost task 0.0 in stage 3180.0 (TID 12048, localhost, executor driver): com.pingcap.tikv.codec.InvalidCodecFormatException: Invalid Flag type for TimestampType: 4
	at com.pingcap.tikv.types.TimestampType.decodeNotNull(TimestampType.java:62)
	at com.pingcap.tikv.types.DataType.decodeValueToRow(DataType.java:133)
	at com.pingcap.tikv.row.DefaultRowReader.readRow(DefaultRowReader.java:38)
	at com.pingcap.tikv.operation.iterator.CoprocessIterator$1.next(CoprocessIterator.java:78)
	at com.pingcap.tikv.operation.iterator.CoprocessIterator$1.next(CoprocessIterator.java:74)
	at org.apache.spark.sql.tispark.TiRDD$$anon$2.next(TiRDD.scala:79)
	at org.apache.spark.sql.tispark.TiRDD$$anon$2.next(TiRDD.scala:55)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.agg_doAggregateWithKeys$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIterator.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$8$$anon$1.hasNext(WholeStageCodegenExec.scala:377)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:126)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:96)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.scheduler.Task.run(Task.scala:99)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Driver stacktrace: 

```

Seems we did't handle `UINT_FLAG` type in `TimestampType`

**Comment by birdstorm** [comment]
`UINT_FLAG`  is not valid concerning `TimestampType`, we are somewhat retrieving wrong data.

**Comment by birdstorm** [comment]
In Addition:

When you use `group by` along with timestamp, you will receive the same error. The same problem exists in any other aggregation.
```
scala> spark.sql("select * from ts group by ts").show(truncate=false)
```
not sure if it is related to issues in TiKV.

**Comment by Novemser** [comment]
`select  distinct(tp_datetime)  from full_data_type_table  order by tp_datetime`
also throws the same exception

**Comment by birdstorm** [comment]
The cause for this issue is that TiKV returns us `UINTFLAG` rather than `UVARINTFLAG` when we push `timestamp` type down to coprocessor. It has been confirmed to be a normal behavior, so will be fixed in #181.

**Comment by Novemser** [comment]
Fixed.
